components/SubmissionForm.vue

```diff
@@ -142,37 +148,46 @@
 </template>
 
 <script lang="ts" setup>
-import { ref, watch, type Ref, nextTick } from 'vue'
+import { ref, type Ref, watch, nextTick } from 'vue'
 import * as validations from '../utils/formValidations'
 import { useSubmissionStore } from '~/stores/submissionStore'
 import { useLocaleStore } from '~/stores/localeStore'
 
-enum ValidationReference {
-    Location = 'LOCATION',
-    LastName = 'LAST_NAME',
-    FirstName = 'FIRST_NAME',
-    PrimarySpokenLanguage = 'PRIMARY_SPOKEN_LANGUAGE',
-    SecondarySpokenLanguage = 'SECONDARY_SPOKEN_LANGUAGE'
-}
-
 const submissionStore = useSubmissionStore()
 const localeStore = useLocaleStore()
 
-const validationCheckedPreviously: Ref<boolean> = ref(false)
-
+/*
+These values are used in order to make it more readable but also to add
+interactive for the user. Originally the checks are performed after the
+first submission or the first blur event. Then it checks on change so once
+the user enters a valid format the error goes away
+*/
+const googleMapsValidationCheckedPreviously: Ref<boolean> = ref(false)
 const isValidGoogleMapsUrl: Ref<boolean> = ref(true)
+
+const lastNameValidationCheckedPreviously: Ref<boolean> = ref(false)
 const isValidLastName: Ref<boolean> = ref(true)
+
+const firstNameValidationCheckedPreviously: Ref<boolean> = ref(false)
 const isValidFirstName: Ref<boolean> = ref(true)
+
+const primarySpokenLanguageValidationCheckedPreviously: Ref<boolean> = ref(false)
 const isValidPrimarySpokenLanguage: Ref <boolean> = ref(true)
+
+const secondaryLanguageValidationCheckedPreviously: Ref<boolean> = ref(false)
```

Comment:
I wonder if we could simplify this by adding the values to two separate maps: one for `previouslyChecked` and another for `isValid`. @ermish  thoughts?

Reply:
Originally I had an array of objects. If I understood correctly @ermish commented earlier that by doing certain things that way was maybe overcomplicating it. I think based on the same philosophy of hardcoding the html if it doesn't need to be dynamic for the readability. But let me know if I misunderstood. I quoted his comment from before.

Reply:
@NabbeunNabi I was referring to something different like just a simple map to store the booleans. For example:

```
const checkedPreviously = {
"googleMaps": Ref<boolean> = ref(false), 
"lastName" : Ref<boolean> = ref(false)
}

const isValid = {
"googleMapsUrl": Ref<boolean> = ref(true),
"LastName": Ref<boolean> = ref(true)
}
```

I haven't personally tried using a ref instead of a map so I'm not sure if it would work. It would help to improve the readability though.

Reply:
Great Idea. It should work!